Content.Client/Doors/DoorSystem.cs

```diff
@@ -1,28 +1,132 @@
 using Content.Shared.Doors.Components;
 using Content.Shared.Doors.Systems;
+using Robust.Client.Animations;
 using Robust.Client.GameObjects;
+using Robust.Client.ResourceManagement;
 using Robust.Shared.Audio;
 using Robust.Shared.GameObjects;
 using Robust.Shared.Player;
+using Robust.Shared.Timing;
 
 namespace Content.Client.Doors;
 
 public sealed class DoorSystem : SharedDoorSystem
 {
-    // Gotta love it when both the client-side and server-side sprite components both have a draw depth, but for
-    // whatever bloody reason the shared component doesn't.
-    protected override void UpdateAppearance(EntityUid uid, DoorComponent? door = null)
+    [Dependency] private readonly AnimationPlayerSystem _animationSystem = default!;
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly IResourceCache _resourceCache = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<DoorComponent, AppearanceChangeEvent>(OnAppearanceChange);
+    }
+
+    protected override void OnComponentInit(EntityUid uid, DoorComponent comp, ComponentInit args)
+    {
+        comp.OpenSpriteStates = new(new (DoorVisualLayers, string)[1]{(DoorVisualLayers.Base, comp.OpenSpriteState)});
+        comp.ClosedSpriteStates = new(new (DoorVisualLayers, string)[1]{(DoorVisualLayers.Base, comp.ClosedSpriteState)});
```

Comment:
can you initialize these more explicitly, its a little hard to read

Reply:
Done.